plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/WooPayments.php

```diff
@@ -196,11 +196,25 @@ public function get_onboarding_url( WC_Payment_Gateway $payment_gateway, string
 			$live_onboarding = true;
 		}
 
+		$transient_key = 'wc_experiment_failure_woocommerce_payment_settings_onboarding_2025_v1';
+
+		// Try to get cached result first.
+		$cached_result = get_transient( $transient_key );
+
 		// We run an experiment to determine the efficiency of test-account-first onboarding vs straight-to-live onboarding.
 		// If the experiment is active and the store is in the treatment group, we will do live onboarding.
 		// Otherwise, we will do test-account-first onboarding (control group).
-		if ( ! $live_onboarding && Experimental_Abtest::in_treatment( 'woocommerce_payment_settings_onboarding_2025_v1' ) ) {
-			$live_onboarding = true;
+		try {
+			// If we have a cache entry that indicates an error, disable the live onboarding.
+			if ( 'error' === $cached_result ) {
+				$live_onboarding = false;
+			} else {
+				$live_onboarding = ! $live_onboarding && Experimental_Abtest::in_treatment( 'woocommerce_payment_settings_onboarding_2025_v1' );
```

Comment:
Won't this will disable live onboarding for merchants **who are eligible** for it but are not in the experiment's treatment group? You probably didn't intend that?

Reply:
The earlier check was setting `$live_onboarding` to true in a specific condition. In the new check, `$live_onboarding` may be set as true or **false**. I think that's a side effect when `$live_onboarding` is already true because a merchant meets the eligibility criteria in: https://github.com/woocommerce/woocommerce/blob/4e99a8bfd252ce5881fd6c72372a3e4583298a1d/plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/WooPayments.php#L180-L197.

Reply:
@anu-rock is right on this one. If something fails, we should remember the failure and decide _nothing_ - whatever was decided before, goes. If our routing logic decided it should do live onboarding, so be it; if it decided to do test account onboarding, the same.

To make things clearer, I refactored a little bit and adjusted the code [here](https://github.com/woocommerce/woocommerce/pull/56174/commits/5b75de3160c8cb61a237309ce40257b0ceaaef3c). I also made sure that we don't read the transient unnecessarily (when live onboarding is true).

Reply:
That makes sense! Thank you for working on it @vladolaru 🙇